src/Entity/FileEntity.php

```diff
@@ -83,6 +83,16 @@ public function preSave(EntityStorageInterface $storage) {
     if ($this->bundle() === FILE_TYPE_NONE) {
       $this->updateBundle();
     }
+    // If the original file entity is empty.
```

Comment:
and the comment should then also explain what we are actually doing if that condition is true: Updating the original entity to have the same bundle, so that the field values can be compared.